File: Java/Sorting/insertion_sort.java
public class insertion_sort
{
    public static void sort(int a[],int n) {
         for (int j = 1; j < n; j++) {  
            int k = a[j];  
            int i = j-1;  
            while ( (i > -1) && ( a [i] > k ) ) {  
                a [i+1] = a [i];  
                i--;  
            }  
            a[i+1] = k;  
        }  
        for(int i:a){    
            System.out.print(i+" ");    
        }    
        System.out.println(); 
    }

	public static void main(String[] args) {
		int[] arr = {45, 3, 2, 89,77};
         int n = arr.length;
         sort(arr, n);  
	}
}

File: Python/Sorting/insertion_sort.py
def Sort(a, n):
    for i in range(1, n):
        k = a[i]
        j = i - 1
        
        while j>=0 and a[j] > k:
            a[j+1] = a[j]
            j = j-1
        a[j+1] = k
        
    for i in range(0, n):
        print(a[i], " ")
        
        
arr = [45, 55, 20, 3, 5]
n = len(arr)
Sort(arr, n)